s2spy/time.py

```diff
@@ -304,3 +296,237 @@ def show(self) -> pd.DataFrame:
         return self._label_targets(self.get_intervals()).applymap(
             self._interval_as_weeknr
         )
+
+
+class CustomCalendar(BaseCalendar):
+    """Build a calendar from sratch with basic construction elements."""
+
+    def __init__(self, anchor: Tuple[int, int] = (11, 30)):
+        """Instantiate a basic container for building calendar using basic blocks.
+
+        This is a highly flexible calendar which allows the user to build their own
+        calendar with the basic building blocks of target and precursor periods.
+
+        Note that two rules are applied to avoid information leakage and disorder:
+            - period block next to the anchor date should not have negative gap
+            - if a appended period block has negative gap, its absolute value must
+             be smaller than the length of the precedent period block
+        """
+        self.month = anchor[0]
+        self.day = anchor[1]
+        self._list_target = []
+        self._list_precursor = []
+        self._total_length_target = 0
+        self._total_length_precursor = 0
+        self.n_targets = 0
+        # identifier for completion of appending
+        self._end_indentifier = False
+
+        self._allow_overlap: bool = False
+
+    def _get_anchor(self, year: int) -> pd.Timestamp:
+        """Generates a timestamp for the end of interval 0 in year.
+
+        Args:
+            year (int): anchor year for which the anchor timestamp should be generated
+
+        Returns:
+            pd.Timestamp: Timestamp at the end of the anchor_years interval 0.
+        """
+        return pd.Timestamp(year, self.month, self.day)
+
+    def append(self, period_block):
+        """Append target/precursor periods to the calendar."""
+        if self._end_indentifier:
+            warnings.warn(
+                "Calerdar already exists and this overwrites the old calendar."
+            )
```

Comment:
Is this warning necessary? If you change the calendar settings the intervals can easily be regenerated.

After calling map_years or map_to_data the minimum/maximum extents are stored in the calendar already, so the calendar can be generated after that. Even the order `initialize -> map -> append periods -> show()` should be no problem.

Reply:
Yep, indeed it will not cause trouble to the workflow. Only in case the user accidently calls the appending function after the calendar is already generated. But I agree, it seems more logical that a user calls it on purpose. Let's remove this warning.